go/logic/topology_recovery.go

```diff
@@ -1346,11 +1358,16 @@ func CheckAndRecover(specificInstance *inst.InstanceKey, candidateInstanceKey *i
 			// force mode. Keep it synchronuous
 			var topologyRecovery *TopologyRecovery
 			recoveryAttempted, topologyRecovery, err = executeCheckAndRecoverFunction(analysisEntry, candidateInstanceKey, true, skipProcesses)
+			log.Errore(err)
 			if topologyRecovery != nil {
 				promotedReplicaKey = topologyRecovery.SuccessorKey
 			}
 		} else {
-			go executeCheckAndRecoverFunction(analysisEntry, candidateInstanceKey, false, skipProcesses)
+			analysisEntry := analysisEntry
```

Comment:
@nickvanw thank you; I became accustomed to this paradigm following https://golang.org/doc/effective_go.html#channels (`req := req`).
I thought at first it looked funny, but then it is also very explicit and clear.